diff -ur xfce4-diskperf-plugin-2.3.0-orig/configure.ac xfce4-diskperf-plugin-2.3.0/configure.ac
--- xfce4-diskperf-plugin-2.3.0-orig/configure.ac	2011-01-21 02:03:08.000000000 +0100
+++ xfce4-diskperf-plugin-2.3.0/configure.ac	2011-03-01 12:50:28.232889237 +0100
@@ -54,7 +54,7 @@
 dnl *** Check for required packages ***
 dnl ***********************************
 XDT_CHECK_PACKAGE([LIBXFCE4PANEL], [libxfce4panel-1.0], [4.3.90])
-XDT_CHECK_PACKAGE([LIBXFCEGUI4], [libxfcegui4-1.0], [4.3.90])
+XDT_CHECK_PACKAGE([LIBXFCE4UI], [libxfce4ui-1], [4.8.0])
 
 dnl ***********************************
 dnl *** Check for debugging support ***
diff -ur xfce4-diskperf-plugin-2.3.0-orig/panel-plugin/config_gui.c xfce4-diskperf-plugin-2.3.0/panel-plugin/config_gui.c
--- xfce4-diskperf-plugin-2.3.0-orig/panel-plugin/config_gui.c	2011-01-19 01:25:09.000000000 +0100
+++ xfce4-diskperf-plugin-2.3.0/panel-plugin/config_gui.c	2011-03-01 12:51:20.519889237 +0100
@@ -32,7 +32,7 @@
 #include <gtk/gtk.h>
 
 #include <libxfce4util/libxfce4util.h>
-#include <libxfcegui4/dialogs.h>
+#include <libxfce4ui/libxfce4ui.h>
 
 
 #define COPYVAL(var, field)	((var)->field = field)
diff -ur xfce4-diskperf-plugin-2.3.0-orig/panel-plugin/config_gui.h xfce4-diskperf-plugin-2.3.0/panel-plugin/config_gui.h
--- xfce4-diskperf-plugin-2.3.0-orig/panel-plugin/config_gui.h	2011-01-19 01:25:14.000000000 +0100
+++ xfce4-diskperf-plugin-2.3.0/panel-plugin/config_gui.h	2011-03-01 12:50:57.187889236 +0100
@@ -24,7 +24,7 @@
 #include <gtk/gtk.h>
 
 #include <libxfce4util/libxfce4util.h>
-#include <libxfcegui4/dialogs.h>
+#include <libxfce4ui/libxfce4ui.h>
 
 
 typedef struct gui_t {
diff -ur xfce4-diskperf-plugin-2.3.0-orig/panel-plugin/main.c xfce4-diskperf-plugin-2.3.0/panel-plugin/main.c
--- xfce4-diskperf-plugin-2.3.0-orig/panel-plugin/main.c	2011-01-19 01:25:30.000000000 +0100
+++ xfce4-diskperf-plugin-2.3.0/panel-plugin/main.c	2011-03-01 12:52:20.736889238 +0100
@@ -25,8 +25,8 @@
 #include <gtk/gtk.h>
 
 #include <libxfce4util/libxfce4util.h>
-#include <libxfcegui4/libxfcegui4.h>
-#include <libxfce4panel/xfce-panel-plugin.h>
+#include <libxfce4ui/libxfce4ui.h>
+#include <libxfce4panel/libxfce4panel.h>
 
 #include <stdlib.h>
 #include <unistd.h>
@@ -630,7 +630,7 @@
 
     status = stat (pcDevice, &oStat);
     if (status == -1) {
-	xfce_err ("%s\n"
+	xfce_dialog_show_error (NULL, NULL, "%s\n"
 		  "%s: %s (%d)",
 		  PLUGIN_NAME, pcDevice, strerror (errno), errno);
 	return;
@@ -852,7 +852,7 @@
 	return (0);
     if (status < 0) {
 	status *= -1;
-	xfce_err (_("%s\n"
+	xfce_dialog_show_error (NULL, NULL, _("%s\n"
 		  "%s: %s (%d)\n\n"
 		  "This monitor will not work!\n"
 		  "Please remove it."),
@@ -862,14 +862,14 @@
     }
     switch (status) {
 	case NO_EXTENDED_STATS:
-	    xfce_err (_("%s: No disk extended statistics found!\n"
+	    xfce_dialog_show_error (NULL, NULL, _("%s: No disk extended statistics found!\n"
 		      "Either old kernel (< 2.4.20) or not\n"
 		      "compiled with CONFIG_BLK_STATS turned on.\n\n"
 		      "This monitor will not work!\n"
 		      "Please remove it."), PLUGIN_NAME);
 	    break;
 	default:
-	    xfce_err (_("%s: Unknown error\n\n"
+	    xfce_dialog_show_error (NULL, NULL, _("%s: Unknown error\n\n"
 		      "This monitor will not work!\n"
 		      "Please remove it."), PLUGIN_NAME);
     }
@@ -881,7 +881,7 @@
 static void About (Widget_t w, void *unused)
 	/* Called back when the About button in clicked */
 {
-    xfce_info (_("%s %s - Disk Performance Monitor\n"
+    xfce_dialog_show_info(NULL, NULL, _("%s %s - Disk Performance Monitor\n"
 	       "Display instantaneous disk I/O transfer rates and busy times "
 	       "on Linux and NetBSD systems\n\n"
 	       "(c) 2003, 2004 Roger Seguin <roger_seguin@msn.com>\n"
diff -ur xfce4-diskperf-plugin-2.3.0-orig/panel-plugin/Makefile.am xfce4-diskperf-plugin-2.3.0/panel-plugin/Makefile.am
--- xfce4-diskperf-plugin-2.3.0-orig/panel-plugin/Makefile.am	2011-01-19 01:24:58.000000000 +0100
+++ xfce4-diskperf-plugin-2.3.0/panel-plugin/Makefile.am	2011-03-01 12:50:28.236889237 +0100
@@ -21,11 +21,11 @@
 
 xfce4_diskperf_plugin_CFLAGS =					\
 	$(LIBXFCE4PANEL_CFLAGS)					\
-	$(LIBXFCEGUI4_CFLAGS)
+	$(LIBXFCE4UI_CFLAGS)
 
 xfce4_diskperf_plugin_LDADD=					\
 	$(LIBXFCE4PANEL_LIBS)					\
-	$(LIBXFCEGUI4_LIBS)
+	$(LIBXFCE4UI_LIBS)
 
 #
 # Desktop file
